changelog/new_add_new_internalaffairsmethodnameendwith.md

```diff
@@ -0,0 +1 @@
+* [#10600](https://github.com/rubocop/rubocop/pull/10600): Add new `InternalAffairs/MethodNameEndWith` cop. ([@nobuyo][])
```

Comment:
This is an internal private cop for development. So changelog entry may not be required.